include/guisan/listmodel.hpp

```diff
@@ -92,6 +92,12 @@ namespace gcn
          * @return An element as a string at the a certain index.
          */
         virtual std::string getElementAt(int i) = 0;
+
+        // Add a new element
+        virtual void add(const std::string& str) {}
+
+        // Clear all elements
+        virtual void clear_elements() {}
```

Comment:
Better to have `= 0;` IMO for both.
but maybe inherit from this class would be better.

Reply:
Adapter the code accordingly. Also added = default for the destructor